scipy/stats/_multivariate.py

```diff
@@ -3850,3 +3851,418 @@ def rvs(self, dim, size=1, random_state=None):
+    (``x`` and ``m``) are interpreted as if each row along the last axis
+    is a single object. For instance:
+
+    >>> multivariate_hypergeom.pmf(x=[[3, 4], [4, 5]], m=[5, 10], n=[7, 9])
+    array([0.32634033, 0.25174825])
+
+    Here, ``x.shape == (2, 2)``, ``m.shape == (2,)``, and ``n.shape == (2,)``,
+    but following the rules mentioned above they behave as if the rows
+    ``[3, 4]`` and ``[3, 5]`` in ``x``, ``[5, 10]`` in ``m``, and ``[7, 9]``
+    in ``n`` were a single object, and as if we had ``x.shape = (2,)``,
+    ``m.shape = (2,)``, and ``n.shape = ()``. To obtain the individual elements
+    without broadcasting, we would do this:
+
+    >>> multivariate_hypergeom.pmf([3, 4], m=[5, 10], n=7)
+    0.3263403263403265
+    >>> multivariate_hypergeom.pmf([4, 5], m=[5, 10], n=9)
+    0.25174825174825205
+
+    This broadcasting also works for ``cov``, where the output objects are
+    square matrices of size ``m.shape[-1]``. For example:
+
+    >>> multivariate_hypergeom.cov(m=[[7, 9], [10, 15]], n=[8, 12])
+    array([[[ 1.05, -1.05],
+            [-1.05,  1.05]],
+
+           [[ 1.56, -1.56],
+            [-1.56,  1.56]]])
+
+    In this example, ``m.shape == (2, 2)`` and ``n.shape == (2,)``,
+    broadcast, following the rules above, as if ``m.shape == (2,)``.
+    Thus the result should also be of shape ``(2,)``, but since each output is
+    a :math:`2 \times 2` matrix. The result, in fact, has shape ``(2, 2, 2)``,
+    where ``result[0]`` is equal to ``multivariate_hypergeom.cov(m=[7, 9], n=8)``
+    and ``result[1]`` is equal to ``multivariate_hypergeom.cov(m=[10, 15], n=12)``.
+
+    See also
+    --------
+    scipy.stats.hypergeom
+    numpy.random.Generator.multivariate_hypergeometric
+
+    References
+    ----------
+    .. [1]: The Multivariate Hypergeometric Distribution,
+            http://www.randomservices.org/random/urn/MultiHypergeometric.html
+
+    .. [2]: Thomas J. Sargent and John Stachurski, 2020,
+            Multivariate Hypergeometric Distribution on Wikipedia
+            https://python.quantecon.org/_downloads/pdf/multi_hyper.pdf
+    """
+    def __init__(self, seed=None):
+        super(multivariate_hypergeom_gen, self).__init__(seed)
+        self.__doc__ = doccer.docformat(self.__doc__)
+
+    def __call__(self, m, n, seed=None):
+        """
+        Create a frozen multivariate_hypergeom distribution.
+
+        See `multivariate_hypergeom_frozen` for more information.
+        """
+        return multivariate_hypergeom_frozen(m, n, seed=seed)
+
+    def _process_parameters(self, m, n):
+        m = np.asarray(m, dtype=np.int_)
+        n = np.asarray(n, dtype=np.int_)
+        if m.ndim == 0:
+            raise ValueError("'m' must be an array!")
+        mcond = np.any(m < 0, axis=-1)
+
+        M = m.sum(axis=-1)
+
+        ncond = (n <= 0) | (n > M)
+        return M, m, n, mcond | ncond
+
+    def _process_quantiles(self, x, M, m, n):
```

Comment:
I and @mdhaber had a discussion on broadcasting arrays before running any code so the traceback stack is clear. This is the reason `M` is present in the method. It is not used anywhere (in the computation) except to broadcast all the arrays. The code implemented now is a little obfuscated, so if you have any improvment suggesstion, it will be very helpful!

Reply:
ok, I see `M` being used now, so closing